Video/Video/View Controllers/ViewRecordingViewController.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  ViewRecordingViewController.swift
+//  Video
+//
+//  Created by Wyatt Harrell on 5/6/20.
+//  Copyright © 2020 Wyatt Harrell. All rights reserved.
+//
+
+import UIKit
+import AVFoundation
+import CoreLocation
+
+extension UIViewController {
+    func hideKeyboardWhenViewTapped() {
+        let tap: UITapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(UIViewController.dismissKeyboard))
+        tap.cancelsTouchesInView = false
+        view.addGestureRecognizer(tap)
+    }
+    @objc func dismissKeyboard() {
+        view.endEditing(true)
+    }
+}
+
+extension UITextField {
+    func setLeftPaddingPoints(_ amount:CGFloat){
+        let paddingView = UIView(frame: CGRect(x: 0, y: 0, width: amount, height: self.frame.size.height))
+        self.leftView = paddingView
+        self.leftViewMode = .always
+    }
+    func setRightPaddingPoints(_ amount:CGFloat) {
+        let paddingView = UIView(frame: CGRect(x: 0, y: 0, width: amount, height: self.frame.size.height))
+        self.rightView = paddingView
+        self.rightViewMode = .always
+    }
+}
+
+class ViewRecordingViewController: UIViewController {
+
+    // MARK: - IBOutlets
+    @IBOutlet weak var videoPlayerView: VideoPlayerView!
+    @IBOutlet weak var nameTextField: UITextField!
+    @IBOutlet weak var cancelButton: UIButton!
+    @IBOutlet weak var saveButton: UIButton!
+    @IBOutlet weak var authorTextField: UITextField!
+    
+    // MARK: - Properties
+    private var player: AVPlayer!
+    var locationManager: CLLocationManager?
+    var recordingURL: URL?
+    var latitude: Double = 0
+    var longitude: Double = 0
+    
+    // MARK: - View Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.hideKeyboardWhenViewTapped()
+        updateViews()
+        playMovie(url: recordingURL)
+        
+        NotificationCenter.default.addObserver(forName: .AVPlayerItemDidPlayToEndTime, object: self.player.currentItem, queue: .main) { [weak self] _ in
+            self?.player?.seek(to: CMTime.zero)
+            self?.player?.play()
+        }
+        
+        if CLLocationManager.locationServicesEnabled() {
+            locationManager?.delegate = self
+            locationManager?.desiredAccuracy = kCLLocationAccuracyNearestTenMeters
+            locationManager?.startUpdatingLocation()
+        }
```

Comment:
I think maybe rather than waiting until you get to this screen to start updating the location, it might be a good idea to do this right after you request permission, since if you're in an area where the GPS signal is sketchy it could take awhile to get the location. Or, you could create a separate helper class that handles all of this to abstract that logic out of the view controller. You could even potentially make it a singleton, since really there will only be one location for the device.